docs/api/proposed_endpoints.md

```diff
@@ -507,3 +527,420 @@ Status: 200 OK
+            "item": "12abcff0-bcbc-01fc-abcd-01012bc4e0b1",
+            "description": "sponsered by someone",
+            "image": null,
+            "created_at": "2018-08-01T17:30:42Z",
+            "modified_at": "2018-08-01T17:30:42Z"
+        }
+    ]
+}
+```
+
+## Delete a swag
+
+```
+DELETE /api/swags/:id (requires authentication)
+```
+
+**Response**
+Status: 204 No-Content
+
+## Upload image of a swag
+
+```
+POST /api/swags/:id/upload_image (requires authentication)
+```
+
+Image will be uploaded as multipart data as a streaming HTTP request.
+
+__Note__:
+- `image` key will be used to send multipart data.
+
+**Response**
+Status: 201 Created
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "item": "12abcff0-bcbc-01fc-abcd-01012bc4e0b1",
+    "description": "sponsered by someone",
+    "image": "http://xyz.com/url/of/uploaded_image.jpg",
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z"
+}
+```
+
+## Delete the image of swag
+
+```
+DELETE /api/swags/:id/delete_image (requires authentication)
+```
+
+**Response**
+Status: 200 OK
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "item": "12abcff0-bcbc-01fc-abcd-01012bc4e0b1",
+    "description": "sponsered by someone",
+    "image": null,
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z"
+}
+```
+
+## Create a swag item
+
+**Parameters**
+
+Name              | Data Type | Required | Default Value  | Discription
+------------------|-----------|----------|----------------|--------------------
+id                | UUID      | false    | _uuid_         | Unique ID for the swag item.
+name              | text      | true     | null           | Name of the item.
+created_at        | datetime  | false    | _datetime_     | Time of creation.
+modified_at       | datetime  | false    | _datetime_     | Time of last modification.
+
+__Note__:
+- _uuid_: Randomly generated Universally Unique IDentifier (UUID).
+- _datetime_: Date and time when the `OtherSwag` object is created.
+
+```
+POST /api/swags/items (requires authentication)
+```
+
+**Request**
+```json
+{
+    "name": "charger"
+}
+```
+
+**Response**
+Status: 201 Created
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "name": "charger",
+    "created_at": "2018-08-01T17:30:41",
+    "modified_at": "2018-08-01T17:30:41"
+}
+```
+
+## Update the swag item
+
+```
+PATCH /api/swags/items/:id (requires authentication)
+```
+
+**Request**
+```json
+{
+    "name": "mug",
+    "created_at": "2018-08-01T17:30:41",
+    "modified_at": "2018-08-01T17:30:41"
+}
+```
+
+**Response**
+Status: 200 OK
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "name": "mug",
+    "created_at": "2018-08-01T17:30:41",
+    "modified_at": "2018-08-01T17:30:41"
+}
+```
+
+## Get details of a swag item
+
+```
+GET /api/swags/items/:id (requires authentication)
+```
+
+**Response**
+Status: 200 OK
+```json
+{
```

Comment:
If this is just one item, why not give a name to swag altogether rather than trying to link them through a foreign key?